features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/TimelineItemEventRow.kt

```diff
@@ -637,58 +636,74 @@ private fun ReplyToContent(
                 textContent = eventContent.body,
                 type = AttachmentThumbnailType.Image,
                 blurHash = type.info?.blurhash,
             )
-            is VideoMessageType -> AttachmentThumbnailInfo(
+        )
+        is VideoMessageType -> InReplyToMetadata.Thumbnail(
+            AttachmentThumbnailInfo(
                 thumbnailSource = type.info?.thumbnailSource,
                 textContent = eventContent.body,
                 type = AttachmentThumbnailType.Video,
                 blurHash = type.info?.blurhash,
             )
-            is FileMessageType -> AttachmentThumbnailInfo(
+        )
+        is FileMessageType -> InReplyToMetadata.Thumbnail(
+            AttachmentThumbnailInfo(
                 thumbnailSource = type.info?.thumbnailSource,
                 textContent = eventContent.body,
                 type = AttachmentThumbnailType.File,
             )
-            is LocationMessageType -> AttachmentThumbnailInfo(
-                textContent = eventContent.body,
+        )
+        is LocationMessageType -> InReplyToMetadata.Thumbnail(
+            AttachmentThumbnailInfo(
+                textContent = stringResource(CommonStrings.common_shared_location),
                 type = AttachmentThumbnailType.Location,
             )
-            is AudioMessageType -> AttachmentThumbnailInfo(
+        )
+        is AudioMessageType -> InReplyToMetadata.Thumbnail(
+            AttachmentThumbnailInfo(
                 textContent = eventContent.body,
                 type = AttachmentThumbnailType.Audio,
             )
-            is VoiceMessageType -> AttachmentThumbnailInfo(
+        )
+        is VoiceMessageType -> InReplyToMetadata.Thumbnail(
+            AttachmentThumbnailInfo(
+                textContent = stringResource(CommonStrings.common_voice_message),
                 type = AttachmentThumbnailType.Voice,
             )
-            else -> null
-        }
-        is PollContent -> AttachmentThumbnailInfo(
+        )
+        else -> InReplyToMetadata.Text(textContent ?: eventContent.body)
+    }
+    is PollContent -> InReplyToMetadata.Thumbnail(
+        AttachmentThumbnailInfo(
             textContent = eventContent.question,
             type = AttachmentThumbnailType.Poll,
         )
-        else -> null
-    }
+    )
+    else -> null
 }
 
-@Composable
-private fun textForInReplyTo(inReplyTo: InReplyToDetails): String {
-    return when (val eventContent = inReplyTo.eventContent) {
-        is MessageContent -> when (eventContent.type) {
-            is LocationMessageType -> stringResource(CommonStrings.common_shared_location)
-            is VoiceMessageType -> stringResource(CommonStrings.common_voice_message)
-            else -> inReplyTo.textContent ?: eventContent.body
-        }
-        is PollContent -> eventContent.question
-        else -> ""
-    }
+private sealed interface InReplyToMetadata {
```

Comment:
I don't really know if it actually makes a difference, but using `@Immutable` with sealed interfaces *should* help in avoiding unnecessary recompositions, since otherwise the compose compiler doesn't know how to treat the interface AFAIK.

Reply:
To the `sealed interface`.